hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSEditLog.java

```diff
@@ -1657,16 +1657,11 @@ synchronized void logEdit(final int length, final byte[] data) {
   /**
    * Run recovery on all journals to recover any unclosed segments
    */
-  synchronized void recoverUnclosedStreams() {
+  synchronized void recoverUnclosedStreams() throws IOException {
     Preconditions.checkState(
         state == State.BETWEEN_LOG_SEGMENTS,
         "May not recover segments - wrong state: %s", state);
-    try {
-      journalSet.recoverUnfinalizedSegments();
-    } catch (IOException ex) {
-      // All journals have failed, it is handled in logSync.
-      // TODO: are we sure this is OK?
-    }
+    journalSet.recoverUnfinalizedSegments();
```

Comment:
This looks right to me as we've been discussing, but I would appreciate another pair of eyes on it to see if I'm missing anything. @omalley can you take a look? (see discussion above on why we're making this change)